finality-provider/store/fpstore_test.go

```diff
@@ -173,20 +175,22 @@ func TestUpdateFpStatusFromVotingPower(t *testing.T) {
 	homePath := t.TempDir()
 	cfg := config.DefaultDBConfigWithHomePath(homePath)
 
-	fpdb, err := cfg.GetDbBackend()
+	fpdb, err := cfg.GetDBBackend()
 	require.NoError(t, err)
 	fps, err := fpstore.NewFinalityProviderStore(fpdb)
 	require.NoError(t, err)
 
-	defer func() {
+	t.Cleanup(func() {
 		err := fpdb.Close()
 		require.NoError(t, err)
 		err = os.RemoveAll(homePath)
 		require.NoError(t, err)
-	}()
+	})
 
 	for _, tc := range tcs {
+		tc := tc
```

Comment:
how could this work?

Reply:
This might be a false positive by the linter, but remember in previous versions of golang when running a go routine in a loop you couldn't just pass the loop variable into the go routine but you had to either to this, or pass it as an argument to the go routine